### PR TITLE
feat: create fast api wrapper

### PR DESCRIPTION
**Pull request by jmargaret2**
### Summary
Create stub of FastAPI wrapper for our Gradescope API

### Details
Adding the stubs for the FastAPI wrapper to gain feedback and review of the changes.

### Checks
- [x] Tested changes
- [ ] Attached Logs

### Team to Review
All

### Reference to the issue
_Mention the issue ID or resources_

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def test_update_assignment_due_date():
+    response = client.put(
+        "/assignment/due_date",
+        headers={"Authorization": "Bearer token"},
+        json={"assignment_id": "assignment_id", "due_date": "due_date"},
+    )
+    assert response.status_code == 200
+    assert response.json() == {"message": "Assignment due date updated"}
```
Comment (SushieeK) [inline tests/test_api.py:53]: Tests look good to me.

**Review by SushieeK — COMMENTED** [review]
PR looks good to me, I hope integrating it with the other components will be taken care next. 💯

File: src/sylveon/_config/config.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    access_token: str
+
+
+class CourseModel(BaseModel):
+    course_id: str
+    course_short_name: str
+    course_name: str
+    course_semester: SemesterModel
+    course_year: int
+
```
Comment (KevZhe) [inline src/sylveon/_config/config.py:36]: small thing, courses may also have the attribute ```num_grades_published``` if the course is an instructor's course

File: src/sylveon/_config/config.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class UserModel(BaseModel):
+    user_id: str
```
Comment (KevZhe) [inline src/sylveon/_config/config.py:39]: Does gradescope have user ids/usernames? Or is this just a placeholder for like the user's full name

File: src/sylveon/api/api.py
Comment (KevZhe) [inline src/sylveon/api/api.py:1]: looks good to me as a starting point

File: src/sylveon/_config/config.py (hunk 3 above)
Comment (jmargaret2) [reply src/sylveon/_config/config.py:39]: just acts as a placeholder for the user's full name/whatever identifier Gradescope will use to identify individual users

File: src/sylveon/_config/config.py (hunk 2 above)
Comment (jmargaret2) [reply src/sylveon/_config/config.py:36]: added to PR

**Review by calvinatian — APPROVED** [review]
I'm a little confused by the commit history picking up stuff from other commits and showing the diff here too. When merged to main will this cause duplicate commits or merge conflicts? Can you try merging main to your branch first?

Otherwise lgtm

**Comment by KevZhe** [comment]
yea, what calvin said. we would have to clean up the commit history if it dupes

**Review by KevZhe — APPROVED** [review]

**Pull request merged**